sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -504,6 +504,19 @@ object LikeSimplification extends Rule[LogicalPlan] {
             Like(input, Literal.create(pattern, StringType))
         }
       }
+
+    case Like(Literal(pattern, StringType), input) =>
```

Comment:
Left input to `Like` is string input. This makes confusing.  Right parameter is `pattern`.

```scala
case Like(Literal(input, StringType), pattern) =>
```